kintoclient/__init__.py

```diff
@@ -0,0 +1,344 @@
+        return Collection(name, bucket=self, session=self.session,
+                          create=True, permissions=permissions)
+
+    def delete_collection(self, name):
+        uri = self._get_collection_uri(name)
+        self.session.request('delete', uri)
+
+    def save(self):
+        self.session.request('patch', self.uri, data=self.data,
+                             permissions=self.permissions)
+
+    def delete(self):
+        self.session.request('delete', self.uri)
+
+
+# XXX Refactor the permissions to use dicts + defaultdicts rather
+# than attributes.
+class Permissions(object):
+    """Handles the permissions as sets"""
+    def __init__(self, object, permissions=None):
+        objects = OBJECTS_PERMISSIONS.keys()
+        if object not in objects:
+            msg = 'object should be one of %s' % ','.join(objects)
+            raise AttributeError(msg)
+
+        if permissions is None:
+            permissions = {}
+
+        self.object = object
+        self.permissions = permissions
+
+        for permission_type in OBJECTS_PERMISSIONS[object]:
+            attr = permission_type.replace(':', '_')
+            setattr(self, attr, permissions.get(permission_type, []))
+
+    def as_dict(self):
+        """Serialize the permissions to be sent to the server"""
+        to_save = {}
+        for permission_type in OBJECTS_PERMISSIONS[self.object]:
+            attr = permission_type.replace(':', '_')
+            to_save[permission_type] = list(getattr(self, attr))
+        return to_save
+
+    def __repr__(self):
+        return "<Permissions on %s: %s>" % (self.object, str(self.permissions))
+
+
+class Collection(object):
+    """Represents a collection. A collection is a parent for records, and
+    has permissions attached to it.
+    """
+    def __init__(self, name, bucket='default', permissions=None,
+                 server_url=None, auth=None, session=None, create=False,
+                 load=True):
+        """
+        :param name:
+            The name of the collection.
+        :param server_url:
+            The URL of the server to use.
+        :param auth:
+            A requests authentication policy object.
+        :param bucket:
+            The bucket object which owns this collection.
+        :param session:
+            An optional session object to use, rather than creating a new one.
+        :param create:
+            Defines if the collection should be created. (default to False)
+        :param load:
+            Defines if collection data should be loaded or not
+            (defaults to True)
+        """
+        self.session = create_session(server_url, auth, session)
+        if isinstance(bucket, six.string_types):
+            # XXX refactor the url routing in a router object.
+            bucket = Bucket(bucket, session=self.session, load=False)
+        self.bucket = bucket
+        self.name = name
+        self.uri = "%s/collections/%s" % (self.bucket.uri, self.name)
```

Comment:
At the moment there are no authorized characters that needs to be urlencoded.
See: https://github.com/mozilla-services/cliquet/blob/master/cliquet/storage/generators.py#L14

But if we want to be future proof on that we may want to do so.